Adding  docs for the `k6/experimental/websockets` module (#933)
grafana/k6-docs: src/data/markdown/docs/02 javascript api/07 k6-experimental/03 websockets/10 WebSocket/30-WebSocket-onclose.md

---
title: 'WebSocket.onclose(callback)'
excerpt: 'Set up a callback function for WebSocket connection close event.'
---

Set up a callback function for WebSocket connection `close` event. That way you can set up the only handler for the `close` event. If you want to have simultaneously multiple event handlers, please use [`WebSocket.addEventHandler()`](/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener).

| Parameter | Type     | Description                                  |
| --------- | -------- | -------------------------------------------- |
| callback  | function | The function to call when the event happens. |

### Example

> _A k6 script that will initiate WebSocket connection sends a ping and closes it using `onopen` handler. We also should see `WebSocket connection closed` that will come from `onclose` handler._

Reviewer: ```suggestion
_A k6 script that initiates a WebSocket connection, sends a ping, and closes it using `onopen` handler.
The console should also log `WebSocket connection closed` from the `onclose` handler._
```

---
title: 'WebSocket.onclose'
excerpt: 'A handler function for WebSocket connection close event.'
---

A handler for a WebSocket connection `close` event.
For multiple, simultaneous event handlers, use [`WebSocket.addEventListener()`](/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener).

| Parameter | Type     | Description                                  |
| --------- | -------- | -------------------------------------------- |

> | handler  | function | The function to call when the event happens. |

Reviewer 2: I think we can drop this table for all handler functions, unless the function itself takes a parameter (like `onmessage`), in which case we would describe its parameters.

Author: I agree :+1: 

For that PR, let's maybe drop all such tables since they don't bring any additional value. And for the `onmessage` (and similar handlers) let's document the handler's function signature in the following PRs.
